### PR TITLE
ci: gather logs when deploying Rook fails

### PR DESCRIPTION
**Pull request by nixpanic**
Log a few commands that help troubleshooting Rook deployment issues.
This might need to get extended with more commands.

Updates: #1636

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: scripts/rook.sh
Comment (nixpanic) [inline scripts/rook.sh:17]: Maybe the namespace `rook-ceph` needs to be added?

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:18]: Don't you want the operator logs too?

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:17]: Do you really want  `-e`? I'd continue if anything fails in that function since it's meant to help debugging.

Comment (leseb) [reply scripts/rook.sh:17]: Nevermind! Read too fast!

File: scripts/rook.sh
Comment (nixpanic) [reply scripts/rook.sh:18]: Yes, maybe. Currently most issues are probably due to Docker Pull limits, but the operator logs would be helpful in other cases.

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:11]: If you trap on `ERR` instead you don't need to untrap at the end IIRC.

File: scripts/rook.sh
Comment (Madhu-1) [inline scripts/rook.sh:20]: pod describe the output of pods in the rook-ceph namespace also helps

File: scripts/rook.sh
Comment (nixpanic) [reply scripts/rook.sh:11]: That looks nice, I'll use that.

File: scripts/rook.sh
Comment (nixpanic) [reply scripts/rook.sh:20]: sure, I can do that instead

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:243]: This is not needed anymore since you trap for `ERR` now.

Comment (nixpanic) [reply scripts/rook.sh:243]: duh :man_facepalming:

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:11]: Now you need to use `errtrace` (`-E`) in addition to `errexit` (`-e`) to properly trap `ERR`.

File: scripts/rook.sh
Comment (leseb) [inline scripts/rook.sh:24]: One last thing I guess, don't you want to `exit 1` after showing the logs so the CI fails?

**Review by leseb — CHANGES_REQUESTED** [review]

File: scripts/rook.sh
Comment (nixpanic) [reply scripts/rook.sh:24]: The script runs with `bash -e` so when an error is caught, the script exits with the error value. An additional `exit 1` is not needed (also tested locally with a small script).

File: scripts/rook.sh
Comment (nixpanic) [reply scripts/rook.sh:11]: Not sure, with a small test script this seems to work just fine?
```bash
#!/bin/bash -e

trap log_exit ERR

function log_exit() {
	set -x
	echo "oh no"
}

false

exit 0
```

Comment (leseb) [reply scripts/rook.sh:11]: I think it works because `false` is not in a function. If you have a function that fails this won't work see: https://stackoverflow.com/questions/35800082/how-to-trap-err-when-using-set-e-in-bash

Comment (nixpanic) [reply scripts/rook.sh:11]: Oh, ok! Yes, failing inside a function indeed makes a difference. Will adapt it with `-E` and exit when the logging function gets called (otherwise it runs 2x, it seems).

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@leseb do you have any comments? CI is waiting for your approval

**Review by leseb — APPROVED** [review]

**Pull request merged**
